src/plots/cartesian/graph_interact.js

```diff
@@ -310,27 +310,45 @@ function hover(gd, evt, subplot) {
 
     if(!subplot) subplot = 'xy';
 
+    // if the user passed in an array of subplots,
+    // use those instead of finding overlayed plots
+    var subplots = Array.isArray(subplot) ? subplot : [subplot];
+
     var fullLayout = gd._fullLayout,
-        plotinfo = fullLayout._plots[subplot],
-
-        //If the user passed in an array of subplots, use those instead of finding overlayed plots
-        subplots = Array.isArray(subplot) ?
-            subplot :
-            // list of all overlaid subplots to look at
-            [subplot].concat(plotinfo.overlays
-                .map(function(pi) { return pi.id; })),
-
-        xaArray = subplots.map(function(spId) {
-            var ternary = (gd._fullLayout[spId] || {})._ternary;
-            if(ternary) return ternary.xaxis;
-            return Axes.getFromId(gd, spId, 'x');
-        }),
-        yaArray = subplots.map(function(spId) {
-            var ternary = (gd._fullLayout[spId] || {})._ternary;
-            if(ternary) return ternary.yaxis;
-            return Axes.getFromId(gd, spId, 'y');
-        }),
-        hovermode = evt.hovermode || fullLayout.hovermode;
+        plots = fullLayout._plots || [],
```

Comment:
Mostly for convenience. 

Most of our per-subplot code post-supply-defaults is designed as:

1. loop over all subplots of a given type found `fullLayout` 
2. create a subplot object or update an existing subplot object in `fullLayout[/* name of subplot */]._subplot` 

So, it becomes convenient to store a ref to the subplot object in its corresponding `fullLayout` attribute container.

Moreover, as finding all xy subplots in a given figure is somewhat expensive (see [`Axes.getSubplots`](https://github.com/plotly/plotly.js/blob/476818c5ff6c3c7301bb897c5af13ade83c848d0/src/plots/cartesian/axes.js#L1153)), it becomes advantages to call look for them in `fullLayout._plots` directly instead of calling `Axes.getSubplots` every time we need to loop over all the x/y subplots.